File: src/backend/studio-reservation/src/main/java/org/taonaw/studio_reservation/domain/model/staffAccount/StaffAccountId.java
package org.taonaw.studio_reservation.domain.model.staffAccount;

import de.huxhorn.sulky.ulid.ULID;
import lombok.Getter;
import org.taonaw.studio_reservation.domain.shared.Assertion;

@Getter
public class StaffAccountId {
    private final String value;

    public StaffAccountId(String value) {
        Assertion.required(value);
        this.value = value;
    }

    public static StaffAccountId newId() {
        return new StaffAccountId(new ULID().nextULID());
    }
}

File: src/backend/identityaccess/src/main/kotlin/org/taonaw/identityaccess/domain/model/staffAccount/StaffAccountId.kt
package org.taonaw.identityaccess.domain.model.staffAccount

import de.huxhorn.sulky.ulid.ULID

data class StaffAccountId(val value: String) {
    companion object {
        fun newId(): StaffAccountId {
            return StaffAccountId(ULID().nextULID())
        }
    }
}